ocaml/typing/parmatch.ml

```diff
@@ -1988,9 +1988,12 @@ let do_check_partial ~pred loc casel pss = match pss with
       exhaust None pss (List.length ps)
       |> Seq.filter_map (typecheck ~pred) in
     match counter_examples () with
-    | Seq.Nil -> Total
+    | Seq.Nil ->
+        if not should_be_exhaustive && Warnings.is_active (Warnings.Exhaustive_match)
```

Comment:
```suggestion
        if not should_be_exhaustive && Warnings.is_active Exhaustive_match
```
We could use type-directed disambiguation here and below to avoid stuttering. I'm also fine doing the same mini-change for `Warnings.is_active (Warnings.Partial_match "")` as you're changing that line anyway, so might as well do a small cleanup.